tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -461,14 +461,19 @@ def _assert_mqtt_topic_messages(
         minimum: int = 1,
         maximum: int = None,
         message_pattern: str = None,
+        message_contains: str = None,
         **kwargs,
     ) -> List[Dict[str, Any]]:
         # log.info("Checking mqtt messages for topic: %s", topic)
+        if message_contains:
+            message_pattern = re.escape(message_contains)
```

Comment:
Is this done on purpose? i.e. using `message_contain` as a pattern even is a pattern has been explicitly provided?

Reply:
Yes it was done on purpose since generally you only use either a pattern or a contains action. Though yes it could mean that if someone does use both it will "silently" ignore the message_pattern given by the user.